packages/tldraw/src/lib/shapes/shared/useEditableText.ts

```diff
@@ -20,52 +20,22 @@ export function useEditableText<T extends Extract<TLShape, { props: { text: stri
 	const editor = useEditor()
 
 	const rInput = useRef<HTMLTextAreaElement>(null)
-	const rSkipSelectOnFocus = useRef(false)
 	const rSelectionRanges = useRef<Range[] | null>()
 
 	const isEditing = useValue('isEditing', () => editor.getEditingShapeId() === id, [editor, id])
-	const shape = editor.getShape(id) as T
-	const doesShapeAutoEditOnKeystroke =
-		shape && editor.getShapeUtil(type).doesAutoEditOnKeyStroke(shape)
 
 	// If the shape is editing but the input element not focused, focus the element
 	useEffect(() => {
 		const elm = rInput.current
 		if (elm && isEditing && document.activeElement !== elm) {
 			elm.focus()
 		}
-
-		// Place the cursor at the end of the text.
-		if (elm && isEditing && doesShapeAutoEditOnKeystroke) {
-			elm.setSelectionRange(elm.value.length, elm.value.length)
-		}
-	}, [isEditing, doesShapeAutoEditOnKeystroke])
+	}, [isEditing])
 
 	// When the label receives focus, set the value to the most  recent text value and select all of the text
 	const handleFocus = useCallback(() => {
-		if (doesShapeAutoEditOnKeystroke) {
-			return
-		}
-
-		// Store and turn off the skipSelectOnFocus flag
-		const skipSelect = rSkipSelectOnFocus.current
-		rSkipSelectOnFocus.current = false
-
-		// On the next frame, if we're not skipping select AND we have text in the element, then focus the text
-		requestAnimationFrame(() => {
-			const elm = rInput.current
-			if (!elm) return
-
-			const shape = editor.getShape<TLShape & { props: { text: string } }>(id)
-
-			if (shape) {
-				elm.value = shape.props.text
-				if (elm.value.length && !skipSelect) {
-					elm.select()
-				}
-			}
-		})
-	}, [editor, id, doesShapeAutoEditOnKeystroke])
+		/* do nothing */
+	}, [])
```

Comment:
can this be deleted entirely? if it's a required prop somewhere, you could pass in `noop`

Reply:
sorry, this hasn't been synced in a while with its base PR (it's a stacked PR) — this section hasn't been deleted.

i'll actually unstack this, because it's confusing now.

Reply:
actually, i retract that comment - i _did_ need this to be removed for all this to work, i started losing track of all of this 🙃 

`noop` sounds good!